bindings/wasm/src/common/timestamp.rs

```diff
@@ -30,10 +31,69 @@ impl WasmTimestamp {
   pub fn to_rfc3339(&self) -> String {
     self.0.to_rfc3339()
   }
+
+  /// Computes `self + duration`
+  ///
+  /// Returns `null` if the operation leads to a timestamp not in the valid range for [RFC 3339](https://tools.ietf.org/html/rfc3339).
+  #[wasm_bindgen(js_name = checkedAdd)]
+  pub fn checked_add(self, duration: WasmDuration) -> Option<WasmTimestamp> {
+    self.0.checked_add(duration.0).map(WasmTimestamp)
+  }
+
+  /// Computes `self - duration`
+  ///
+  /// Returns `null` if the operation leads to a timestamp not in the valid range for [RFC 3339](https://tools.ietf.org/html/rfc3339).
+  #[wasm_bindgen(js_name = checkedSub)]
+  pub fn checked_sub(self, duration: WasmDuration) -> Option<WasmTimestamp> {
+    self.0.checked_sub(duration.0).map(WasmTimestamp)
+  }
 }
 
 impl From<Timestamp> for WasmTimestamp {
   fn from(timestamp: Timestamp) -> Self {
     Self(timestamp)
   }
 }
+
+/// A span of time.
+#[wasm_bindgen(js_name = Duration, inspectable)]
+pub struct WasmDuration(pub(crate) Duration);
+
+#[wasm_bindgen(js_class = Duration)]
+impl WasmDuration {
+  /// Create a new `Duration` with the given number of seconds.
+  pub fn seconds(seconds: u32) -> WasmDuration {
+    Self(Duration::seconds(seconds))
+  }
+  /// Create a new `Duration` with the given number of minutes.
+  pub fn minutes(minutes: u32) -> WasmDuration {
+    Self(Duration::minutes(minutes))
+  }
+
+  /// Create a new `Duration` with the given number of hours.
+  pub fn hours(hours: u32) -> WasmDuration {
+    Self(Duration::hours(hours))
+  }
+
+  /// Create a new `Duration` with the given number of days.
+  pub fn days(days: u32) -> WasmDuration {
+    Self(Duration::days(days))
+  }
+
+  /// Create a new `Duration` with the given number of weeks.
+  pub fn weeks(weeks: u32) -> WasmDuration {
+    Self(Duration::weeks(weeks))
+  }
```

Comment:
I think it works regardless but we should place `#[wasm_bindgen]` annotations on exported functions.